xds/src/main/java/io/grpc/xds/XdsServerWrapper.java

```diff
@@ -559,7 +559,9 @@ public void run() {
             if (!routeDiscoveryStates.containsKey(resourceName)) {
               return;
             }
-            savedVirtualHosts.set(ImmutableList.copyOf(update.virtualHosts));
+            if (savedVirtualHosts.getAndSet(ImmutableList.copyOf(update.virtualHosts)) == null) {
```

Comment:
We don't want _any_ log statements if everything works properly. We only want this log statement if previously it was a failure. Maybe check `isPending == false` as well?